### PR TITLE
Respond with HTTP status 412 on failure (fixes #8)

### LINKED ISSUE #8: `autospec-web` should somehow allow to error out when curling for the error messages.
It should be possible to query `autospec-web` while getting a non-zero exit code for failures.

The naive `curl localhost:8080` does not work since `curl` always returns `0` when the server responds. My next try was `curl --fail localhost:8080`, which should translate non-200 http status codes to exit code `22`. But that doesn't help since `autospec-web` always returns 200.

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: src/Run.hs [hunk 1]
```diff
@@ -57,10 +59,10 @@ runWeb args = do
...
+trigger :: Session -> IO (Bool, String)
 trigger interpreter = do
   xs <- Interpreter.reload interpreter
-  ys <- if "Ok, modules loaded:" `isInfixOf` xs
+  (ys, summary) <- if "Ok, modules loaded:" `isInfixOf` xs
     then Interpreter.hspec interpreter
-    else return ""
-  return (xs ++ ys)
+    else return ("", Nothing)
+  return (maybe False ((== 0) . summaryFailures) summary, xs ++ ys)
```
Comment (soenkehahn) [inline src/Run.hs]: So the summary may be `Nothing` when parsing the `Summary` fails? And we interpret that as failure?

File: src/Interpreter.hs [hunk 2]
```diff
@@ -32,5 +39,20 @@ close = GhciWrapper.close . sessionInterpreter
...
+  r <- evalEcho sessionInterpreter $ "System.Environment.withArgs " ++ (show $ "--color" : sessionHspecArgs) ++ " $ Test.Hspec.Runner.hspecResult spec"
+  return $ case reverse $ lines r of
+    x : _ | Just summary <- readMaybe (dropAnsiEscapeSequences x) -> (r, Just summary)
+    _ -> (r, Nothing)
+  where
+    dropAnsiEscapeSequences xs
+      | "Summary" `isPrefixOf` xs = xs
+      | otherwise = case xs of
+          _ : ys -> dropAnsiEscapeSequences ys
+          [] -> []
```
Comment (soenkehahn) [inline src/Interpreter.hs]: Why is this needed? Why do we drop the first character when the string doesn't start with `"Summary"`?

**Comment by soenkehahn** [comment]
Apart from comments (and travis) LGTM.

File: src/Interpreter.hs (hunk 2 above)
Comment (jkarni) [inline src/Interpreter.hs]: fwiw, it drops _all_ characters up to `Summary`

Comment (soenkehahn) [inline src/Interpreter.hs]: Yes, we talked about this offline. `hspec`s output includes the closing escape sequence (to reset to the default color) after the newline of the previous line. So it has to be stripped here. And just dropping everything till the sequence `"Summary"` seems to be the easiest.

**Pull request merged**